app/ui/legacy/src/main/java/com/fsck/k9/ui/settings/SettingsViewModel.kt

```diff
@@ -2,24 +2,22 @@ package com.fsck.k9.ui.settings
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.asLiveData
+import androidx.lifecycle.viewModelScope
 import com.fsck.k9.Account
 import com.fsck.k9.preferences.AccountManager
 import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 
 internal class SettingsViewModel(
     private val accountManager: AccountManager,
-    private val coroutineScope: CoroutineScope = GlobalScope,
     private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO,
 ) : ViewModel() {
     val accounts = accountManager.getAccountsFlow().asLiveData()
 
     fun moveAccount(account: Account, newPosition: Int) {
-        coroutineScope.launch(coroutineDispatcher) {
+        viewModelScope.launch(coroutineDispatcher) {
```

Comment:
We probably want to add a `withContext(NonCancellable)` here. 

As a user I would expect the account reordering to be persisted once the action has been triggered, even if I navigate away from the screen before the animation is done.